solutions/day1.py

```diff
@@ -0,0 +1,45 @@
+
+def parse_input(filename):
+    list1, list2 = [], []
+
+    with open(filename, "r") as f:
+        for line in f:
+            nums =  [int(num) for num in line.split()]
+            list1.append(nums[0])
+            list2.append(nums[1])
+    
+    return list1, list2
+    
+
+def question_1(list1, list2):
+
+    list1, list2 = sorted(list1), sorted(list2)
+    
+    diff = 0
+    for i in range(len(list1)):
+        diff += abs(list1[i] - list2[i])
+    
+    return diff
+
+def question_2(list1, list2):
+
+    right_freq = {}
+
+    for val in list2:
+        if val in right_freq:
+            right_freq[val] += 1
+        else:
+            right_freq[val] = 1
+    
+    similarity_score = 0
+    for num in list1:
+        if num in right_freq:
+            similarity_score += right_freq[num] * num
```

Comment:
It's a little faster and not much harder to read if you run through the key, values in `right_freq`; there are no/fewer lookups and the length of the key set in `right_freq` could be much shorter than `len(list1)`.  
```suggestion
    for k, v in right_freq.items():
        similarity_score += k * v
```